geode-core/src/test/java/org/apache/geode/management/internal/configuration/validators/RegionConfigValidatorTest.java

```diff
@@ -33,10 +37,20 @@
 
   @Before
   public void before() throws Exception {
-    validator = new RegionConfigValidator();
+    InternalCache cache = mock(InternalCache.class);
+    when(cache.getSecurityService()).thenReturn(SecurityServiceFactory.create());
+    validator = new RegionConfigValidator(cache);
     config = new RegionConfig();
   }
 
+  @Test
+  public void checkSecurityForDiskAccess() {
+    config.setName("regionName");
+    config.setType(RegionType.REPLICATE_PERSISTENT);
+    validator.validate(config);
+    assertThat(config.getType()).isEqualTo("REPLICATE_PERSISTENT");
```

Comment:
should we add a verify here?

Reply:
@jinmeiliao I'm not sure what you mean exactly.

Reply:
I thought the reason you are adding this test is to verify that authorize method is called, otherwise simply asserting that the type is the same as the what's set doesn't seem very useful.

So probably we should create a mock SecurityService and verify that authorize method is called with the correct parameter like this:
```
public void before(){
....
securityService = mock(SecurityService.class);
when(cache.getSecurityService()).thenReturn(securityService);
....
}

@Test
  public void checkSecurityForDiskAccess() {
    ...
    validator.validate(config);
    verify(securityService).authorize(ResourcePermission.Resource.CLUSTER, ResourcePermission.Operation.WRITE, ResourcePermission.Target.DISK);
    ....
  }

  @Test
  public void noChangesWhenTypeIsSet() {
    ...
    validator.validate(config);
    verify(securityService, times(0)).authorize(any());
    ....
  }

```